src/common/transformations/src/transformations/common_optimizations/common_optimizations.cpp

```diff
@@ -215,6 +216,7 @@ bool ngraph::pass::CommonOptimizations::run_on_model(const std::shared_ptr<ngrap
     // other optimizations
     manager.register_pass<ngraph::pass::StridesOptimization>();
     manager.register_pass<ngraph::pass::Validate>();
+    manager.register_pass<ov::pass::ResolveGeneratedNameCollisions>();
```

Comment:
Plugins usually call some transformations after CommonOptimizations. What if we get a name conflict there?

Reply:
In this case plugins should update their pipelines.

Reply:
So should they register ResolveGeneratedNameCollisions once again after all transformations?

Reply:
Good point. I looked into initial problem, and looks like that only GPU plugin suffer from the same names for different nodes. May be it is even better to enable this transformation only for GPU plugin in the end of its transformation pipeline.

Reply:
Ok I updated the GPU pipeline